learning/bootstrap.py

```diff
@@ -109,11 +109,27 @@ async def teacher_loop(cfg: DictConfig, mle_log: MLELogger):
 
             # terminate the learning loop if all final goals are proven
             if len(success_logprobs_final) == len(final_goals):
+                mean_success_logprobs_final = sum(success_logprobs_final)/len(success_logprobs_final)
                 log.info('All final goals proven - stopping learning loop...')
                 mle_log.update({'num_iterations': i},
+                            'mean_success_logprobs_final': mean_success_logprobs_final,
                            {'final_goals_proven': final_goals_proven})
                 break
 
+            # get logprobs of proving the final goals (with far more mcts steps) as a progress_metric for us
+            search_budget = min(cfg.agent.max_mcts_nodes*100, 10000)
+            while len(success_logprobs_final) == 0 and search_budget < 100000:
+                student_results_final = prove_conjectures(agent_dump, final_goals_formatted, theory, premises, search_budget)
+                success_logprobs_final = get_log_probs(student_results_final, i)
+                search_budget *= 2
```

Comment:
I'm not sure about the exponential backoff logic here. We should either constrain the MCTS search to timeout after `search_budget` steps, or we let the MCTS search run indefinitely, i.e. until proving the theorem.

Currently we would be re-running MCTS from zero when the exponential backoff logic is triggered, thus wasting a lot of compute.